mslib/msui/linearview.py

```diff
@@ -232,3 +234,191 @@ def open_settings_dialog(self):
             settings = dlg.get_settings()
             self.getView().plotter.set_settings(settings, save=True)
         dlg.destroy()
+
+    def get_settings(self):
+        """Return a dictionary of all linear view settings."""
+        # Get settings from the view (matplotlib canvas)
+        view_settings = self.getView().get_settings()
+
+        # Get WMS settings
+        wms_settings = {}
+        if self.docks and self.docks[0] is not None:
+            wms_settings = {
+                "url": self.currurl,
+                "layer": self.currlayer,
+                "level": self.currlevel,
+                "styles": self.currstyles,
+                "flights": self.currflights,
+                "vertical": self.currvertical,
+            }
+
+        # Get dock widget states
+        dock_states = [dock is not None for dock in self.docks]
+
+        return {
+            "view_type": "linearview",
+            "plot_title_size": view_settings.get("plot_title_size", "10pt"),
+            "axes_label_size": view_settings.get("axes_label_size", "10pt"),
```

Comment:
the default can be taken from the config_loader, because a user could have changed them.